libvast/src/view.cpp

```diff
@@ -204,9 +204,27 @@ bool evaluate_view(const data_view& lhs, relational_operator op,
   };
   auto check_in = [](const auto& x, const auto& y) {
     return caf::visit(detail::overload(
-                        [](auto, auto) {
-                          // Default case.
-                          return false;
+                        [](auto lhs, auto rhs) {
+                          if constexpr (detail::is_any_v<decltype(rhs),
+                                                         view<vector>,
+                                                         view<set>>) {
+                            auto equals_lhs = [&](auto y) {
+                              if constexpr (std::is_same_v<decltype(lhs),
+                                                           decltype(y)>)
+                                return lhs == y;
+                              else
+                                return false;
+                            };
+                            auto pred = [&](const auto& rhs_element) {
+                              return caf::visit(equals_lhs, rhs_element);
+                            };
+                            return std::find_if(rhs->begin(), rhs->end(), pred)
+                                   != rhs->end();
+                            return false;
+                          } else {
+                            // Default case.
+                            return false;
+                          }
```

Comment:
It's really hard to read this crammed code with so many line breaks. Mind factoring the visitor into a separate function?